zilliqa/src/consensus.rs

```diff
@@ -1921,40 +1916,18 @@ impl Consensus {
         Ok(())
     }
 
-    pub fn leader_at_block(&self, block: &Block, view: u64) -> Option<Validator> {
+    pub fn leader_at_block(&self, block: &Block) -> Option<Validator> {
         let Ok(state_at) = self.try_get_state_at(block.number()) else {
             return None;
         };
-        Some(self.leader(&state_at, view))
-    }
 
-    pub fn leader(&self, state: &State, view: u64) -> Validator {
-        let committee = state.get_stakers().unwrap();
+        let public_key = state_at.leader_at_block(block).unwrap();
```

Comment:
Here we should pass in the `view` instead of the `block`, which is not used in `State::leader_at_block()`.